ios/PurchasesHybridCommon/PurchasesHybridCommon/RCPurchaserInfo+HybridAdditions.m

```diff
@@ -7,48 +7,64 @@
+        allPurchases[productIdentifier] = [self formattedAsISO8601OrNull:date];
+        allPurchasesMillis[productIdentifier] = [self timeIntervalSince1970OrNull:date];
     }
 
-    id latestExpiration = self.latestExpirationDate.formattedAsISO8601 ?: [NSNull null];
-
     return @{
-            @"entitlements": self.entitlements.dictionary,
-            @"activeSubscriptions": self.activeSubscriptions.allObjects,
-            @"allPurchasedProductIdentifiers": self.allPurchasedProductIdentifiers.allObjects,
-            @"latestExpirationDate": latestExpiration,
-            @"latestExpirationDateMillis": self.latestExpirationDate ? @(self.latestExpirationDate.timeIntervalSince1970) : [NSNull null],
-            @"firstSeen": self.firstSeen.formattedAsISO8601,
-            @"firstSeenMillis": @(self.firstSeen.timeIntervalSince1970),
-            @"originalAppUserId": self.originalAppUserId,
-            @"requestDate": self.requestDate.formattedAsISO8601,
-            @"requestDateMillis": @(self.requestDate.timeIntervalSince1970),
-            @"allExpirationDates": allExpirations,
-            @"allExpirationDatesMillis": allExpirationsMillis,
-            @"allPurchaseDates": allPurchases,
-            @"allPurchaseDatesMillis": allPurchasesMillis,
-            @"originalApplicationVersion": self.originalApplicationVersion ?: [NSNull null],
+        @"entitlements": self.entitlements.dictionary,
+        @"activeSubscriptions": self.activeSubscriptions.allObjects,
+        @"allPurchasedProductIdentifiers": self.allPurchasedProductIdentifiers.allObjects,
+        @"latestExpirationDate": [self formattedAsISO8601OrNull:self.latestExpirationDate],
+        @"latestExpirationDateMillis": [self timeIntervalSince1970OrNull:self.latestExpirationDate],
+        @"firstSeen": self.firstSeen.formattedAsISO8601,
+        @"firstSeenMillis": @(self.firstSeen.timeIntervalSince1970),
+        @"originalAppUserId": self.originalAppUserId,
+        @"requestDate": self.requestDate.formattedAsISO8601,
+        @"requestDateMillis": @(self.requestDate.timeIntervalSince1970),
+        @"allExpirationDates": allExpirations,
+        @"allExpirationDatesMillis": allExpirationsMillis,
+        @"allPurchaseDates": allPurchases,
+        @"allPurchaseDatesMillis": allPurchasesMillis,
+        @"originalApplicationVersion": self.originalApplicationVersion ?: NSNull.null,
+        @"originalPurchaseDate": [self formattedAsISO8601OrNull:self.originalPurchaseDate],
+        @"originalPurchaseDateMillis": [self timeIntervalSince1970OrNull:self.originalPurchaseDate],
     };
 }
 
+- (NSObject *)timeIntervalSince1970OrNull:(nullable NSDate *)date {
+    if (date) {
+        return @(date.timeIntervalSince1970);
+    } else {
+        return NSNull.null;
+    }
+}
+
+- (NSObject *)formattedAsISO8601OrNull:(nullable NSDate *)date {
+    if (date) {
+        return date.formattedAsISO8601;
+    } else {
+        return NSNull.null;
+    }
+}
```

Comment:
Should we move these two to `NSDate+HybridAdditions.h`?

Reply:
that wouldn't work, unfortunately, because if the date is nil the func will no-op. it could be a static method, but I'm not sure it would add much value

Reply:
oh got it